### PR TITLE
Add ignoreProperties: [] to number-max-precision

### LINKED ISSUE #5420: Add ignoreProperties: [] to number-max-precision
> What is the problem you're trying to solve?

Hi, I've tried to apply this rule to our project and found at least one valid usage of values with decimal parts in pixels — letter-spacing property. So I propose to add ignoreProperties option to this rule.

### PR DESCRIPTION
**Pull request by eljusto**
> Which issue, if any, is this issue related to?

Resolved #5420 

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

**Review by jeddy3 — CHANGES_REQUESTED** [review]
@eljusto Thank you for the pull request. Adding this option SGTM.

Can you please:

1. set the pull request to merge into the `v14` branch
2. add the option to the rule README ([like so](https://stylelint.io/user-guide/rules/list/property-no-vendor-prefix/#ignoreproperties-regex-regex-string))

File: lib/rules/number-max-precision/README.md [hunk 1]
```diff
@@ -46,6 +46,30 @@ a { top: 3.24px; }
 
 ## Optional secondary options
 
+### `ignoreProperties: ["/regex/", /regex/, "string"]`
+
+Ignore the precision of numbers for the specified properties.
+
```
Comment (jeddy3) [inline lib/rules/number-max-precision/README.md:52]:
```suggestion

For example, with `0`.

```

We'll need to give the primary option.

File: lib/rules/number-max-precision/README.md [hunk 2]
```diff
@@ -46,6 +46,30 @@ a { top: 3.24px; }
...
+```
+["transition"]
+```
+
+The following patterns are considered violations:
+
+<!-- prettier-ignore -->
+```css
+a { -o-transition: all 4.5s ease; }
```
Comment (jeddy3) [inline lib/rules/number-max-precision/README.md:63]:
```suggestion
a { top: 10.5px; }
```

Let's use a more common property as some users may not be familiar with the old Opera prefix.

**Review by jeddy3 — CHANGES_REQUESTED** [review]
Thanks for making the changes. I've requested two more changes as suggestions.

File: lib/rules/number-max-precision/README.md [hunk 3]
```diff
@@ -46,6 +46,32 @@ a { top: 3.24px; }
...
 
+### `ignoreProperties: ["/regex/", /regex/, "string"]`
+
+Ignore the precision of numbers for the specified properties.
+
+For example, with `0`.
+
+Given:
+
+```
```
Comment (ybiquitous) [inline lib/rules/number-max-precision/README.md:57]: [trivial] Specifying a language for the code block seems better to me:

````suggestion
```json
````

**Review by ybiquitous — APPROVED** [review]
Thank you. LGTM 👍🏼

**Review by jeddy3 — APPROVED** [review]
Thanks!

**Comment by jeddy3** [comment]
v14 changelog entry:

> - Added: `ignoreProperties: []` to `number-max-precision` ([#5421](https://github.com/stylelint/stylelint/pull/5421)).

**Pull request merged**